components/dm/spec/grafana.go

```diff
@@ -145,6 +149,28 @@ func (i *GrafanaInstance) InitConfig(e executor.Executor, clusterName, clusterVe
 	return e.Transfer(fp, dst, false)
 }
 
+func (i *GrafanaInstance) initDashboards(e executor.Executor, spec GrafanaSpec, paths meta.DirPaths) error {
+	dashboardsDir := filepath.Join(paths.Deploy, "dashboards")
+	// To make this step idempotent, we need cleanup old dashboards first
+	if _, stderr, err := e.Execute(fmt.Sprintf("rm -f %s/*", dashboardsDir), false); err != nil {
+		return errors.Annotatef(err, "cleanup old dashboards: %s", string(stderr))
+	}
+
+	if spec.DashboardDir != "" {
+		if err := i.TransferLocalConfigDir(e, spec.DashboardDir, dashboardsDir); err != nil {
+			return errors.Annotate(err, "transfer dashboards failed")
+		}
+		return nil
+	}
+
+	// Use the default ones
```

Comment:
Why the default files have higher priority?

Reply:
The default ones will be used only in the case `spec.DashboardDir == ""`, otherwise this function will return early

Reply:
Sorry, ignore me.